app/src/main/java/com/example/speakout/ui/activity/LoginActivity.kt

```diff
@@ -1,11 +1,50 @@
 package com.example.speakout.ui.activity
 
+import android.content.Intent
 import android.os.Bundle
+import android.view.View
+import android.widget.Toast
 import androidx.appcompat.app.AppCompatActivity
+import com.example.speakout.LoginContract
+import com.example.speakout.LoginPresenter
+import com.example.speakout.R
+import kotlinx.android.synthetic.main.activity_login.*
 
-class LoginActivity: AppCompatActivity() {
+class LoginActivity: AppCompatActivity(), LoginContract.View{
+
+    private val presenter = LoginPresenter(this)
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_login)
+        setClickListeners()
+    }
+
+    private fun setClickListeners() {
```

Comment:
I prefered not to use it because of this method does not require to "buttton.setOnClickListener(this)" and then I would need to see wich view was clicked.

Reply:
ok